### PR TITLE
[@types/chrome] Update RuleCondition from type to interface

### PR DESCRIPTION
**Pull request by zuhairtaha**
The `chrome.declarativeNetRequest.RuleCondition` has been updated from a type to an interface to resolve an issue with [tsickle](https://www.npmjs.com/package/tsickle) during the conversion of TypeScript types into JSDoc for the Closure Compiler externs. This change also better reflects the intended usage of the [RuleCondition ](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#type-RuleCondition)as an interface.

For example the following TypeScript type:

```ts
declare type T = {
  a?: boolean;
  b?: string[];
} & ({ c?: number[] } | { d?: string[] });
```
Will be converted by [tsickle](https://www.npmjs.com/package/tsickle) to this externs:

```js
/** @typedef {?} */
var T;
```

However, by using an interface instead, like this:
```ts
declare interface T {
  a?: boolean;
  b?: string[];
  c?: number[];
  d?: string[];
}
```

The generated externs will be more detailed and accurate:
```js
/**
 * @record
 * @struct
 */
function T() {}
/** @type {(undefined|boolean)} */
T.prototype.a;
/** @type {(undefined|!Array<string>)} */
T.prototype.b;
/** @type {(undefined|!Array<number>)} */
T.prototype.c;
/** @type {(undefined|!Array<string>)} */
T.prototype.d;
```

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ramda ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ramda@latest/))
was missing the following properties:
1. modifyPath
1. sequence
1. unwind
1. whereAny
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3eb0d466c669b6f1e64793a2d023c8d233bb7894
</p>

**Comment by zuhairtaha** [comment]
@DangerBotOSS What are these properties you mentioned. I cannot find them in `index.d.ts` or in [RuleCondition description](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#type-RuleCondition)

**Comment by zuhairtaha** [comment]
Ready to merge

File: types/chrome/index.d.ts [hunk 1]
```diff
@@ -12013,26 +12013,15 @@ declare namespace chrome.declarativeNetRequest {
...
         urlFilter?: string | undefined;
-    } & (
-            | {
-                /**
-                 * List of resource types which the rule won't match.
-                 * Only one of {@link chrome.declarativeNetRequest.RuleCondition.resourceTypes}
-                 * and {@link chrome.declarativeNetRequest.RuleCondition.excludedResourceTypes} should be specified.
-                 * If neither of them is specified, all resource types except "main_frame" are blocked.
-                 */
-                excludedResourceTypes?: ResourceType[] | undefined;
```
[the remark this answers is not in the document]
Comment (zuhairtaha) [reply types/chrome/index.d.ts:12024]: If you look at the interface `RuleCondition` it also includes `excludedResourceTypes` field and all fields match docs

**Review by gabritto — APPROVED** [review]

**Comment by zuhairtaha** [comment]
Kan you check please
@matthewkimber, @otiai10, @sreimer15, @matcarlson, @ekinsol, @EchoAbstract, @spasma, @bdbai, @pokutuna, @JasonXian, @userTim, @idan315, @nicolas377, @idosal

**Comment by matcarlson** [comment]
@zuhairtaha - FYI, you can also comment Ready to Merge and it'll merge/release it.

**Pull request merged**